util/patch/utils.go

```diff
@@ -45,3 +46,17 @@ func unstructuredHasStatus(u *unstructured.Unstructured) bool {
 	_, ok := u.Object["status"]
 	return ok
 }
+
+func toUnstructured(obj runtime.Object) (*unstructured.Unstructured, error) {
```

Comment:
Can you add a unit test that verifies that when you call toUnstructured with both kinds of objects (unstructured & not) and you then mutate what toUnstructured returns, the original objects are not mutated?

Reply:
Yes

Reply:
All done!